src/useDeepLinking.ts

```diff
@@ -0,0 +1,39 @@
+import {Linking} from 'react-native'
+import {useRef, useEffect} from 'react'
+
+interface handlerParam {
+  url: string
+}
+
+/**
+ * @param deepLinkingHandler Function called when application is opened by a deep link
+ * @param normalHandler Function called when application is NOT opened by a deep link
+ */
+export const useDeepLinking = (
+  deepLinkingHandler: (url: string) => any,
+  normalHandler?: () => any,
+) => {
+  const savedCallback = useRef<any>()
+
+  useEffect(() => {
+    savedCallback.current = deepLinkingHandler
+
+    Linking.getInitialURL().then((url) => {
+      if (url) {
+        deepLinkingHandler(url)
+      } else if (normalHandler) {
+        normalHandler()
+      }
+    })
+  }, [deepLinkingHandler, normalHandler])
```

Comment:
This will run `Linking.getInitialURL()` any time that the `deepLinkingHandler` or `normalHandler` is changed, is this intentional? 🤔 

e.g. the example provided in the readme will trigger the handler on every re-render

Reply:
Ohh, I didn't seem to get that kind of behavior, but no, that is not intentional. The `useDeepLinking()` hook should trigger the `deepLinkingHandler` or/and the `normalHandler` when the app opens. I'm 100% certain on how hooks work, would you be able to direct me into how to fix it?

Thanks in advance and sorry for the later response